### ISSUE TITLE
RAW IP (7) encapsulation not handled by dpkt (-d 3)

### ISSUE THREAD

**Issue by pspacek** [traffic-taffy/traffic-taffy#7]
Versions:
traffic-taffy                 0.9.3
dpkt                          1.9.8
Python 3.12.5

Problem:
PCAPs with Raw IP (7) encapsulation (produced by `dnscap -w`) are not handled by dpkt parser (`-d 3`) and this is not detected. It causes all sort of weird effects from nonsense results to dpkt exceptions being thrown at the user.

Reproducer:
- [bisect10pkts.pcap.gz](https://github.com/user-attachments/files/16836063/bisect10pkts.pcap.gz)
- `taffy-dissect bisect10pkts.pcap`
```
dpkt.dpkt.NeedData: got 34, 35 needed at least
```

Limiting input to `-n1` seemingly allows the file to parse but results are nonsense. Here it shows Ethernet addresses which do not exist in the file at all:
```
$ taffy-dissect -n1 bisect10pkts.pcap
INFO      :	processing bisect10pkts.pcap
Ethernet_dst                   60:06:07:00:00:32              1
Ethernet_dst                   __WIDTH__                      1
Ethernet_src                   11:fa:4d:d8:b0:de              1
Ethernet_src                   __WIDTH__                      1
Ethernet_type                  57779                          1
Ethernet_type                  __WIDTH__                      1
__TOTAL__                      packet                         1
__TOTAL__                      __WIDTH__                      1
```

**Comment by hardaker** [comment]
Fixed in 0.9.4

Note: dpkt doesn't do ipv6 yet, which I need to fix still.

**Comment by hardaker** [comment]
IPv6 should now work too.
